src/CognitiveServices/CognitiveServices/CognitiveServicesAccount/SetAzureCognitiveServicesAccount.cs

```diff
@@ -74,6 +85,51 @@ public class SetAzureCognitiveServicesAccountCommand : CognitiveServicesAccountB
         [AllowEmptyCollection]
         public string CustomSubdomainName { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Generate and assign a new Cognitive Services Account Identity for this storage account for use with key management services like Azure KeyVault.")]
+        public SwitchParameter AssignIdentity { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = false)]
+        [ValidateNotNullOrEmpty]
+        public IdentityType? IdentityType { get; set; }
+
+        [Parameter(HelpMessage = "List of User Owned Storage Accounts.", Mandatory = false)]
+        [ValidateNotNull]
+        [AllowEmptyCollection]
+        public string[] StorageAccountIds { get; set; }
+
+        [Parameter(HelpMessage = "Whether to set Cognitive Services Account Encryption KeySource to Microsoft.CognitiveServices or not.", Mandatory = false, ParameterSetName = CognitiveServicesEncryptionParameterSet)]
+        public SwitchParameter CognitiveServicesEncryption { get; set; }
+
+        [Parameter(HelpMessage = "Whether to set Cognitive Services Account encryption keySource to Microsoft.KeyVault or not. " +
+            "If you specify KeyName, KeyVersion and KeyVaultUri, Cognitive Services Account Encryption KeySource will also be set to Microsoft.KeyVault weather this parameter is set or not.",
+            Mandatory = false, ParameterSetName = KeyVaultEncryptionParameterSet)]
+        public SwitchParameter KeyVaultEncryption { get; set; }
```

Comment:
These two switch params are used to distinguish param sets, so they are **mandatory** in each param set's perspective.

Reply:
What we are trying to do is exact same as StoargeAccount: https://github.com/Azure/azure-powershell/blob/master/src/Storage/Storage.Management/StorageAccount/SetAzureStorageAccount.cs

CognitiveServicesEncryption and KeyVaultEncryption can both be false, if I make them mandatory then at least one of them has to be true.

Reply:
Makes sense